packages/client/components/ShareTopicModal.tsx

```diff
@@ -15,46 +16,188 @@ import {
+            onCompleted()
+            setSelectedIntegration('slack')
+          }
+        })
+      }
+    }
+  }
+
+  const onChannelChange = (channel: string) => {
+    setSelectedChannel(channel)
+  }
+
+  const comingSoonBadge = (
+    <div className='flex items-center justify-center rounded-full bg-slate-300 px-3 py-1'>
+      <div className='text-center text-xs font-semibold text-slate-600'>coming soon</div>
+    </div>
+  )
+
+  const connectButton = (
+    <div className='flex cursor-pointer items-center justify-center rounded-full border bg-white px-3 py-1'>
+      <div className='text-center text-xs font-semibold text-slate-700'>connect</div>
+    </div>
+  )
+
+  const labelStyles = `w-[110px] text-left text-sm font-semibold`
+  const fieldsetStyles = `mx-0 mb-[15px] mb-2 flex items-center gap-5 p-0`
 
   return (
     <Dialog isOpen={isOpen} onClose={onClose}>
-      <DialogContent>
+      <DialogContent className='z-10'>
         <DialogTitle>Share topic</DialogTitle>
         <DialogDescription>Where would you like to share the topic to?</DialogDescription>
 
-        <fieldset className='mx-0 mb-[15px] mb-2 flex items-center gap-5 p-0'>
-          <label className={labelStyles} htmlFor='name'>
-            Integration
-          </label>
-          <Select>
+        <fieldset className={fieldsetStyles}>
+          <label className={labelStyles}>Integration</label>
+          <Select
+            onValueChange={onIntegrationChange}
+            value={selectedIntegration}
+            disabled={submitting}
+          >
             <SelectTrigger>
-              <SelectValue placeholder='Select one' />
+              {selectedIntegration !== '' ? (
+                <SelectValue />
+              ) : (
+                <span className='text-slate-600'>Select one</span>
+              )}
             </SelectTrigger>
             <SelectContent>
               <SelectGroup>
-                <SelectItem value='slack'>Slack</SelectItem>
-                <SelectItem value='teams'>Teams</SelectItem>
-                <SelectItem value='mattermost'>Mattermost</SelectItem>
+                <SelectItem value='slack' endAdornment={!isSlackConnected ? connectButton : null}>
+                  Slack
+                </SelectItem>
+                <SelectItem
+                  value='teams'
+                  disabled={true}
+                  endAdornment={comingSoonBadge}
+                  className='data-[disabled]:opacity-100'
+                >
+                  Teams
+                </SelectItem>
+                <SelectItem
+                  value='mattermost'
+                  disabled={true}
+                  endAdornment={comingSoonBadge}
+                  className='data-[disabled]:opacity-100'
+                >
+                  Mattermost
+                </SelectItem>
               </SelectGroup>
             </SelectContent>
           </Select>
         </fieldset>
 
+        <fieldset className={fieldsetStyles}>
```

Comment:
-1 I think this menu needs more box shadow and maybe more left padding so that it looks similar to our other Slack channels menu:
![Screenshot 2023-07-04 at 15 57 25](https://github.com/ParabolInc/parabol/assets/39854876/a02dafd4-d917-41c9-ab58-2cc9a6d039c9)
![Screenshot 2023-07-04 at 16 01 43](https://github.com/ParabolInc/parabol/assets/39854876/87de20b7-de51-48ba-90fa-97eb4e28383f)

Reply:
@nickoferrall I took padding from the new designs for select component #8232. Should we specifically change it for this particular select? 

For shadow, I'm using tailwind classes instead of defining specific values. For example `shadow-md` https://tailwindcss.com/docs/box-shadow

As we want to migrate to tailwind, I suggest I create a separate issue, to define our own shadows parameters, and keep it as is for now

What do you think?

cc @acressall

Reply:
@igorlesnenko you can start introducing custom box shadow here - https://github.com/ParabolInc/parabol/blob/6a66e4d480837402090e08a7a7e8fe4b6b6b1450/tailwind.config.js#L24

that said, we should be consistent and use predefined values (either from tailwind defaults or our own)

Reply:
@BartoszJarocki thanks! I see there is already defined something, will use that card-1

Reply:
Shadow updated 
<img width="412" alt="image" src="https://github.com/ParabolInc/parabol/assets/466991/b543da09-71be-4b80-b92c-97bffda577d9">

Reply:
Make sure to check with @acressall if the value is correct. Would be nice to start filling the config with proper values 🙏

Reply:
Box shadow is looking good to me! I think there is more left padding though in the other Slack menu, so I think this should be added

![Screenshot 2023-07-05 at 14 46 54](https://github.com/ParabolInc/parabol/assets/39854876/6b7fe8af-0f7f-48f8-b7ac-49423c60951d)

![Screenshot 2023-07-04 at 16 01 43](https://github.com/ParabolInc/parabol/assets/39854876/87de20b7-de51-48ba-90fa-97eb4e28383f)

Reply:
[Here's the design file for drop shadows](https://www.figma.com/file/UAT8GVK86KVKdycX8xtK3tmM/UI-%2F-elevation?type=design&node-id=0-1&mode=design&t=hOhQcMPyxF3CEqlO-0). 
This component uses `1dp` which equates to `box-shadow: 0px 1px 3px 0px rgba(68,66,88,0.45);`